### PR TITLE
feat: multi-hop quotes wired up to trade input

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Wires up swapper quote api to multi-hop trade input with initial implementation of fee summaries. 

Removes cross-account support of lifi trades due to it using the receive address as the sending address during quotes.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

NA

## Risk

Very low risk to lifi swapper trades.

## Testing

Requires quick test of lifi trades.

### Engineering

See above

### Operations

See testing notes

## Screenshots (if applicable)

completed lifi trade on single-hop
image

completed 1inch trade on single-hop
image

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4676** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4676" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4676?utm_source=stack-comment).

File: src/lib/swapper/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts [hunk 1]
```diff
@@ -67,7 +68,7 @@ export async function getTradeQuote(
       toChainId: Number(fromChainId(buyAsset.chainId).chainReference),
       fromTokenAddress: getLifiEvmAssetAddress(sellAsset),
       toTokenAddress: getLifiEvmAssetAddress(buyAsset),
-      fromAddress: receiveAddress,
+      fromAddress: sendAddress,
```
Comment (woodenfurniture) [inline src/lib/swapper/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts:71]: this likely breaks lifi in prod - need to check hence draft PR

File: src/components/Trade/hooks/useTradeQuoteService.tsx [hunk 2]
```diff
@@ -51,19 +50,22 @@ export const useTradeQuoteService = () => {
   useEffect(() => {
     if (sellAsset && buyAsset && wallet && sellAmountCryptoPrecision && sellAccountMetadata) {
       ;(async () => {
-        const { chainId: receiveAddressChainId } = fromAssetId(buyAsset.assetId)
-        const chainAdapter = getChainAdapterManager().get(receiveAddressChainId)
-
-        if (!chainAdapter)
-          throw new Error(`couldn't get chain adapter for ${receiveAddressChainId}`)
```
Comment (woodenfurniture) [inline src/components/Trade/hooks/useTradeQuoteService.tsx:58]: removed this as it has no effect - buy assets are known to be supported

File: src/components/MultiHopTrade/MultiHopTrade.tsx [hunk 3]
```diff
@@ -32,12 +41,14 @@ export const MultiHopTrade = (props: CardProps) => {
   const {
     state: { wallet },
   } = useWallet()
+  const [showTradeQuotes, handleToggleShowTradeQuotes] = useToggle(false)
```
Comment (0xApotheosis) [inline src/components/MultiHopTrade/MultiHopTrade.tsx:44]: "Toggle" is already a verb, so we can probably drop the "handle" whilst still meeting convention.

```suggestion
  const [showTradeQuotes, toggleShowTradeQuotes] = useToggle(false)
```

File: src/components/MultiHopTrade/MultiHopTrade.tsx [hunk 4]
```diff
@@ -86,6 +99,44 @@ export const MultiHopTrade = (props: CardProps) => {
     })
   }, [assetSearch, setBuyAsset, supportedBuyAssets])
 
+  const totalProtocolFees = useMemo(() => {
+    if (!quoteData) return {}
+    return quoteData.steps.reduce<Record<AssetId, ProtocolFee>>((acc, step) => {
```
Comment (0xApotheosis) [inline src/components/MultiHopTrade/MultiHopTrade.tsx:104]: This logic feels better placed in a helper, like we have for `getTotalNetworkFeeFiatPrecision`, for e.g.

File: src/components/MultiHopTrade/MultiHopTrade.tsx [hunk 5]
```diff
@@ -86,6 +99,44 @@ export const MultiHopTrade = (props: CardProps) => {
...
+        },
+        acc,
+      )
+    }, {})
+  }, [quoteData])
+
+  const buyAmountAfterFeesCryptoPrecision = useMemo(() => {
+    if (!quoteData) return '0'
+    const lastStep = quoteData.steps[quoteData.steps.length - 1]
+    const buyAssetProtocolFeesTotal = bnOrZero(
```
Comment (0xApotheosis) [inline src/components/MultiHopTrade/MultiHopTrade.tsx:125]:
```suggestion
    const buyAssetProtocolFeesTotalBaseUnit = bnOrZero(
```

**Review by 0xApotheosis — COMMENTED** [review]
Functionally untested. Some surface level commentary, though otherwise looks good.

File: src/components/MultiHopTrade/components/TradeQuotes/helpers.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  selectFeeAssetById,
+  selectMarketDataByFilter,
+} from 'state/slices/selectors'
+import { store } from 'state/store'
+import { sumProtocolFeesToDenom } from 'state/zustand/swapperStore/utils'
+
+/**
+ * Computes the total receive amount across all hops after protocol fees are deducted
+ * @param param0.quote The trade qutoe
+ * @param param0.swapperName The swapper name
```
[the remark this answers is not in the document]
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/TradeQuotes/helpers.ts:18]: this makes sense intuitively although there is only 1 param here. jsdoc convention is to use dot notation to label properties of params
https://jsdoc.app/tags-param.html

File: src/components/MultiHopTrade/components/TradeQuotes/TradeQuote.tsx [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+  const focusColor = useColorModeValue('blackAlpha.400', 'whiteAlpha.400')
+
+  const { isTradingActive } = useIsTradingActive()
+
+  const buyAsset = useAppSelector(selectBuyAsset)
+  const sellAsset = useAppSelector(selectSellAsset)
+
+  const sellAmountCryptoPrecision = useAppSelector(selectSellAmountCryptoPrecision)
+
+  const handleSwapperSelection = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/TradeQuotes/TradeQuote.tsx:99]: agree, ill rename

File: src/components/MultiHopTrade/components/TradeQuotes/helpers.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  const cryptoMarketDataById = selectCryptoMarketData(store.getState())
+
+  const buyAmountCryptoPrecision = fromBaseUnit(
+    quote.steps[0].buyAmountBeforeFeesCryptoBaseUnit,
+    quote.steps[0].buyAsset.precision,
+  )
+  const slippageDecimalPercentage =
+    quote?.recommendedSlippage ?? getDefaultSlippagePercentageForSwapper(swapperName)
+
+  const buyAmountCryptoPrecisionAfterSlippage = bnOrZero(buyAmountCryptoPrecision)
```
[the remark this answers is not in the document]
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/TradeQuotes/helpers.ts:40]: yeah nice

File: src/components/MultiHopTrade/components/TradeQuotes/TradeQuote.tsx
[the remark this answers is not in the document]
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/TradeQuotes/TradeQuote.tsx:1]: yeah ignoring the fact the old swapper will be deleted, the logic is so different the only reusable part would be the JSX which is subject to total rewrite.

File: src/components/MultiHopTrade/components/TradeQuotes/helpers.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (woodenfurniture) [reply src/components/MultiHopTrade/components/TradeQuotes/helpers.ts:23]: although param0 seems kinda boilerplatey. args.quote etc might work?
also typo to fix

[the remark this answers is not in the document]
Comment (0xApotheosis) [reply src/components/MultiHopTrade/components/TradeQuotes/helpers.ts:23]: Ah sure, so it is 👌

**Review by 0xApotheosis — APPROVED** [review]
Feedback addressed, architecture sane, runtime implications contained behind a flag.

Good to go 🚢

**Pull request merged**